src/vos/vos_tree.c

```diff
@@ -953,10 +979,10 @@ key_tree_punch(struct vos_object *obj, daos_handle_t toh, daos_iov_t *key_iov,
 	/* PROBE_EQ == insert in this case */
 	rc = dbtree_upsert(toh, BTR_PROBE_EQ, DAOS_INTENT_PUNCH, key_iov,
 			   val_iov);
-	if (rc)
+	if (rc != 0 || replay)
```

Comment:
Hmm, you do not need register record for record rebuild?

Reply:
The DTX logic will not affect the modification triggered by rebuild.

Reply:
ah, indeed.